src/main/java/org/apache/lucene/index/KNNMergeHelper.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.apache.lucene.index;
+
+public class KNNMergeHelper {
+
+    private KNNMergeHelper() {}
+    public static boolean isMergeAborted() {
```

Comment:
Can we generalize this to include shard closed events? See this comment: https://github.com/opensearch-project/OpenSearch/issues/8590#issuecomment-2590059735. Im not sure if its possible or not to look up that kind of information. @shwetathareja do you have any ideas around here?